platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherProbeRegistry.java

```diff
@@ -16,22 +16,27 @@
 
 package org.gradle.internal.watch.registry;
 
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.Nullable;
 import java.io.File;
+import java.nio.file.Path;
+import java.util.function.BiConsumer;
 import java.util.stream.Stream;
 
 /**
  * A registry of watch probes. A probe serves as a way to prove that a hierarchy Gradle is interested in
  * indeed receives file system events from the operating system.
  * This is to avoid trusting locations where OSs silently not send any events, despite watchers being registered.
  *
- * When the hierarchy is first registered via {@link #registerProbe(File)}, we don't yet create the probe.
+ * When the hierarchy is first registered via {@link #registerProbe(File, File)}, we don't yet create the probe.
  * That only happens when the hierarchy is actually read or written by Gradle, in which case
- * {@link #armWatchProbe(File)} is called.
+ * {@link org.gradle.internal.watch.registry.impl.DefaultFileWatcherProbeRegistry.WatchProbeImpl#arm()} is called.
```

Comment:
🤔 A reference to a private method here is odd. Consider replacing with just a plain text explanation.